### PR TITLE
CreateCollection: Check if a table follows Data API collection pattern

### PR DESCRIPTION
**Pull request by Yuqi-Du**
When user does createCollection with a collection name called apple, 
table apple exists but that is not a Data API table (column missing), 
we should error out and notify user.

(Previously, since Data API does not error out this situation, and that will give user a weird error with following indexes creation)

**Which issue(s) this PR fixes**:
#1198

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [x] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Review by Hazel-Datastax — COMMENTED** [review]
Possible to add some tests?

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/CreateCollectionOperation.java [hunk 1]
```diff
@@ -327,18 +327,26 @@ public Uni<JsonApiException> cleanUpCollectionFailedWithTooManyIndex(
...
       KeyspaceMetadata currKeyspace,
       String tableName) {
-    // First: do we already have a Table with the same name? If so, return it
+    // First: do we already have a Table with the same name?
     for (TableMetadata table : currKeyspace.getTables().values()) {
       if (table.getName().asInternal().equals(tableName)) {
+        // If that is not a valid Data API table, error out the createCollectionCommand
+        if (!COLLECTION_MATCHER.test(table)) {
+          throw ErrorCode.INVALID_DATA_API_COLLECTION.toApiException(
+              "collection ('%s') already exists but does not follow Data API schema pattern",
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/CreateCollectionOperation.java:346]: Not quite: it is not a Collection by definition (collection is a Table with certain schema).

File: src/main/java/io/stargate/sgv2/jsonapi/exception/ErrorCode.java [hunk 2]
```diff
@@ -75,7 +75,7 @@ public enum ErrorCode {
   SHRED_DOC_LIMIT_VIOLATION("Document size limitation violated"),
 
   EXISTING_COLLECTION_DIFFERENT_SETTINGS("Collection already exists"),
-
+  INVALID_DATA_API_COLLECTION("Invalid Data API collection"),
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/exception/ErrorCode.java:78]: Should this be named similar to above one `EXISTING_COLLECTION_...` as maybe `EXISTING_TABLE`: it is not really a Collection at all but a Table; and we cannot try to re-create it.

**Review by tatu-at-datastax — CHANGES_REQUESTED** [review]
I think we need to think through naming here, if and when adding a new `ErrorCode`
(validation logic itself is fine, only concerned about error reporting).

**Review by vkarpov15 — COMMENTED** [review]
LGTM

**Comment by Yuqi-Du** [comment]
> Possible to add some tests?
I have tested in my local.
This is not a common Data API use case.
In order to replicate we need to create a cassandra table first, and that is a hustle we have never done in Integration Test.

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/collections/CreateCollectionOperation.java [hunk 3]
```diff
@@ -332,18 +332,26 @@ public Uni<JsonApiException> cleanUpCollectionFailedWithTooManyIndex(
...
    */
   TableMetadata findTableAndValidateLimits(
       Map<CqlIdentifier, KeyspaceMetadata> allKeyspaces,
       KeyspaceMetadata currKeyspace,
       String tableName) {
-    // First: do we already have a Table with the same name? If so, return it
+    // First: do we already have a Table with the same name?
     for (TableMetadata table : currKeyspace.getTables().values()) {
       if (table.getName().asInternal().equals(tableName)) {
+        // If that is not a valid Data API table, error out the createCollectionCommand
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/collections/CreateCollectionOperation.java:348]: table -> collection

File: src/main/java/io/stargate/sgv2/jsonapi/exception/ErrorCode.java [hunk 4]
```diff
@@ -82,7 +82,8 @@ public enum ErrorCode {
   SHRED_DOC_LIMIT_VIOLATION("Document size limitation violated"),
 
   EXISTING_COLLECTION_DIFFERENT_SETTINGS("Collection already exists"),
-
+  EXISTING_TABLE_NOT_DATA_API_COLLECTION(
+      "An existing table does not follow Data API collection schema pattern"),
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/exception/ErrorCode.java:86]: nit: An existing -> Existing 
(I am not native speaker and articles are confusing, but here it's a specific table so indefinite "an" seems misplaced)

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/collections/CreateCollectionOperation.java [hunk 5]
```diff
@@ -332,18 +332,26 @@ public Uni<JsonApiException> cleanUpCollectionFailedWithTooManyIndex(
...
+        if (!COLLECTION_MATCHER.test(table)) {
+          throw ErrorCode.EXISTING_TABLE_NOT_DATA_API_COLLECTION.toApiException(
+              "table ('%s') already exists and it does not follow Data API schema pattern",
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/collections/CreateCollectionOperation.java:351]: Maybe "table ('%s') already exists and is not a valid Data API Collection"

File: src/main/java/io/stargate/sgv2/jsonapi/exception/ErrorCode.java [hunk 6]
```diff
@@ -82,7 +82,8 @@ public enum ErrorCode {
   SHRED_DOC_LIMIT_VIOLATION("Document size limitation violated"),
 
   EXISTING_COLLECTION_DIFFERENT_SETTINGS("Collection already exists"),
-
+  EXISTING_TABLE_NOT_DATA_API_COLLECTION(
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/exception/ErrorCode.java:85]: ErrorCode id `EXISTING_TABLE_NOT_DATA_COLLECTION` is ok.

File: src/main/java/io/stargate/sgv2/jsonapi/exception/ErrorCode.java (hunk 4 above)
Comment (tatu-at-datastax) [reply src/main/java/io/stargate/sgv2/jsonapi/exception/ErrorCode.java:86]: Maybe change to: "Existing table not valid Data API Collection"

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/collections/CreateCollectionOperation.java [hunk 7]
```diff
@@ -332,18 +332,26 @@ public Uni<JsonApiException> cleanUpCollectionFailedWithTooManyIndex(
...
 
   /**
-   * Method for finding existing table with given name, if one exists and returning that table; or
-   * if not, verify maximum table limit and return null.
+   * Method for finding existing table with given name: 1. if valid one exists and returning that
+   * table; 2. if invalid one exists, error out; 3. if no table exists with given name, verify
+   * maximum table limit and return null;
    *
-   * @return Existing table with given name, if any; {@code null} if not
+   * @return Existing valid table with given name, if any; {@code null} if not
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/collections/CreateCollectionOperation.java:339]: -> Existing valid Collection

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/collections/CreateCollectionOperation.java [hunk 8]
```diff
@@ -332,18 +332,26 @@ public Uni<JsonApiException> cleanUpCollectionFailedWithTooManyIndex(
   }
 
   /**
-   * Method for finding existing table with given name, if one exists and returning that table; or
-   * if not, verify maximum table limit and return null.
+   * Method for finding existing table with given name: 1. if valid one exists and returning that
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/collections/CreateCollectionOperation.java:335]: table -> collection

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/collections/CreateCollectionOperation.java (hunk 3 above)
Comment (Yuqi-Du) [reply src/main/java/io/stargate/sgv2/jsonapi/service/operation/collections/CreateCollectionOperation.java:348]: solved, thanks

File: src/main/java/io/stargate/sgv2/jsonapi/exception/ErrorCode.java (hunk 4 above)
Comment (Yuqi-Du) [reply src/main/java/io/stargate/sgv2/jsonapi/exception/ErrorCode.java:86]: Got it, solved

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/collections/CreateCollectionOperation.java (hunk 8 above)
Comment (Yuqi-Du) [reply src/main/java/io/stargate/sgv2/jsonapi/service/operation/collections/CreateCollectionOperation.java:335]: solved.

**Review by jeffreyscarpenter — APPROVED** [review]

**Review by tatu-at-datastax — APPROVED** [review]

**Pull request merged**